### DISCUSSION TITLE
Add a decorator to add serialization groups to avoid recursive relation to kicks in when serializing

### DISCUSSION THREAD

**Discussion by SteadEXE** [mikro-orm/mikro-orm#4470] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have a case where I have multiple entities which depends on each others, aka. Parent-Child bidirectionnal relation.
If both side are populated, the serialized entities can get huge.

I come from Symfony world, and we used to get rid of it with serialization groups.

**Describe the solution you'd like**
I would like to have a decorator, to say, okay I only want to serialize those properties when the given group is passed.
class-transformer uses `@Expose()` decorator.
The group can then be passed as an option of `serialize()` function.

**Describe alternatives you've considered**
I was looking to use class-transformer, but it adds dependencies, and also complexity to the project, Mikro ORM is aware of relation, which is a perfect candidate to handle the serialization part more deeper.
I was also trying to implement it in toJSON or toObject, but it quickly get complicated aswell. 
A lot of code get duplicated within the project and accross several projects.

**Additional context**
I tried to find similar issues but I cannot find any, sorry if it has already been asked.

Feel free to reach me if you need more infos.
Have a nice day

**Comment by B4nan** [comment]
I'd say v6 is already dealing with this issue (see #4138). If you want to have absolute control, use explicit serialization.

I am not against more control, but this request is way too sparse, I'd need to see more concrete proposal on how things should work. I am not used to symfony, neither to class-transformer.

**Comment by SteadEXE** [comment]
Sure I understand, I can give you a little bit more context about how it cans look like in Mikro ORM, while explaining how it works in PHP (and also in class-transformer if I remember).

Here is a little snippet:

```typescript
import { Collection, Entity, ManyToOne, OneToMany, PrimaryKey, Property, serialize } from "@mikro-orm/core";

class Parent {
  @PrimaryKey({ autoincrement: true })
  @Expose({ groups: [ 'parents', 'children' ] }) // Proposal @Expose() decorator
  id: number;

  @Property()
  @Expose({ groups: [ 'parents', 'children' ] })
  name: string;

  @Expose({ groups: [ 'parents' ] })
  @OneToMany(() => Child, child => child.parent)
  children = new Collection<Child>(this);
}

@Entity()
class Child {
  @Expose({ groups: [ 'parents', 'children' ] })
  @PrimaryKey({ autoincrement: true })
  id: number;

  @Expose({ groups: [ 'parents', 'children' ] })
  @Property()
  name: string;

  @Expose({ groups: [ 'children' ] })
  @ManyToOne(() => Parent)
  parent!: Parent;
}

const parents = await em
  .getRepository(Parent)
  .find({}, {
    populate: [ 'child', 'child.parent' ] // populate both side to simulate UnitOfWork having everything in memory.
  });

serialize(parents, {
  groups: [ 'parents' ] // Proposal groups options
});

// Without serialization group you will probably end up returning
// Parent (id, name, child)
//    Child (id, name, parent)
//        Parent (id, name, child)
//            Child (id, name, parent)

// With serialization group you will end up
// Parent (id, name, child)
//     Child (id, name) <-- no parent here, because parent is only serialized when using 'children' group.
```

**Reply by SteadEXE** [reply]
With this method, you don't need explicit serialization to exclude everything you don't want, you only get what you want.
It also has the advantage of working without having to exclude the same entity again and again if it appears at different location in your relation tree.

**Reply by B4nan** [reply]
Groups are coming to v6.2, see https://github.com/mikro-orm/mikro-orm/pull/5416
